### PR TITLE
Cleanup scheduler integration tests

### PR DESCRIPTION
**Pull request by ahg-g**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Cleans up the scheduler integration test to use the patterns we settled on as of late.

#### Which issue(s) this PR fixes:

Fixes #

### CODE REVIEW

**Review by denkensk — COMMENTED** [review]
L346 
L356
are also needed to refine.

**Comment by ahg-g** [comment]
> L346 L356 are also needed to refine.

Thanks, done.

**Comment by ahg-g** [comment]
fixed the flake as well: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_kueue/216/pull-kueue-test-integration-main/1514094346826158080

**Comment by denkensk** [comment]
/lgtm

Thanks. @ahg-g

**Comment by alculquicondor** [comment]
> fixed the flake as well

Was the flake only for this PR? or should we cherry-pick to release-0.1?

File: test/integration/framework/framework.go [hunk 1]
```diff
@@ -39,7 +39,7 @@ import (
 	//+kubebuilder:scaffold:imports
 )
 
-type ManagerSetup func(manager.Manager)
+type ManagerSetup func(manager.Manager, context.Context)
```
[the remark this answers is not in the document]
Comment (ahg-g) [reply test/integration/framework/framework.go:42]: to fix the flake.

**Comment by ahg-g** [comment]
No cherrypicking needed, this is a flake caused by the test framework.

**Pull request merged**